### PR TITLE
DAOS-7141 obj: allow size query for EC obj with NULL sgl

### PR DESCRIPTION
**Pull request by liuxuezhao**
One use case in dfs_getxattr() will fetch with NULL sgl to query
iod_size. This patch fix the EC handling for that case.

Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

### CODE REVIEW

File: src/object/cli_ec.c [hunk 1]
```diff
@@ -1520,12 +1520,17 @@ obj_ec_singv_req_reasb(daos_obj_id_t oid, daos_iod_t *iod, d_sg_list_t *sgl,
...
+		if (sgl != NULL) {
+			/* copy the sgl */
+			rc = d_sgl_init(r_sgl, sgl->sg_nr);
+			if (rc)
+				goto out;
+			memcpy(r_sgl->sg_iovs, sgl->sg_iovs,
+			       sizeof(*sgl->sg_iovs) * sgl->sg_nr);
+		} else {
+			r_sgl->sg_iovs = NULL;
+			r_sgl->sg_nr = 0;
```
Comment (wangdi1) [inline src/object/cli_ec.c:1532]: Do we need check sgl for array as well?

**Review by wangdi1 — APPROVED** [review]

**Review by johannlombardi — APPROVED** [review]
It fixes the problem.

File: src/object/cli_ec.c (hunk 1 above)
Comment (liuxuezhao) [reply src/object/cli_ec.c:1532]: I'll add a test case for both singv and array in following patch.

**Comment by liuxuezhao** [comment]
it reports "All tests are passing. Nice one! All 6499 tests for this pipeline are passing."

But also report "test/Functional_Hardware_Medium" failure with err msg as -
[2021-04-04T00:27:03.306Z] Recording test results
[2021-04-04T00:27:03.755Z] [Checks API] No suitable checks publisher found.
Functional_Hardware_Medium failed: 1

**Comment by johannlombardi** [comment]
I see one test failure. Is that known?

**Comment by wangdi1** [comment]
> I see one test failure. Is that known?

yes, DAOS-7179

**Pull request merged**